src/hera/workflows/artifact.py

```diff
@@ -97,6 +108,11 @@ def as_name(self, name: str) -> _ModelArtifact:
         artifact.name = name
         return artifact
 
+    @classmethod
+    def get_input_attributes(cls):
+        """Return the attributes used for input artifact annotations."""
+        return ["name", "path"]
```

Comment:
I think we need to review the input attributes - `from_`, `from_expression` and `artifact_gc` I think should all be possible inputs as well